src/sentry/auth/helper.py

```diff
@@ -41,6 +43,64 @@
 ERR_INVALID_IDENTITY = _('The provider did not return a valid user identity.')
 
 
+class RedisBackedState(object):
```

Comment:
I'm not super happy with how much stuff is going on here, and feel like if we're going to have some type of generic redis session-cache type thing I might as well abstract it into a utility.. maybe I'll do that tomorrow.